### ISSUE TITLE
[IDEA] Better Notification Message

### ISSUE THREAD

**Issue by kookma** [Jermolene/TiddlyWiki5#7341]
Right now, some action messages display a notification like `Copied to clipboard!`. For example

1.permalink
2. permaview
3. ...

This is true for some macros.

Suggestion:

Use better message like `Permalin copied to clipboard!` or allow to customize the message for those action messages.

**Comment by pmario** [comment]
On wikitext level it's [tm-copy-to-clipboard](https://tiddlywiki.com/#WidgetMessage%3A%20tm-copy-to-clipboard) which would need to add an additional parameter

The [notification.multids](https://github.com/pmario/TiddlyWiki5/blob/bbadedce722e87fbfd7961bd6e411e22a1c93ed4/core/language/en-GB/Notifications.multids#L5) is responsible to define this translatable text. ... It would need to use this parameter

The JS code  [.copyToClipboard = function(text,options)](https://github.com/pmario/TiddlyWiki5/blob/bbadedce722e87fbfd7961bd6e411e22a1c93ed4/core/modules/utils/dom/dom.js#L295) would need to use this parameter too. 

There are probably some more js functions which use `.copyToClipboard` .. We need to check them to guarantee backwards compatibility.

**Comment by yaisog** [comment]
Maybe it would help to display the (sanitized) text that was copied, or at least the start thereof? Like
```
Copied to clipboard:
https://www.tiddlywiki.com/#som...
```

**Comment by pmario** [comment]
>Maybe it would help to display the (sanitized) text that was copied, or at least the start thereof? Like

or the password :/

**Comment by Jermolene** [comment]
I agree that it would be useful to add an optional "caption" parameter to the `tm-copy-to-clipboard` macro.

I do also like the suggestion of showing a trimmed version of the copied text in the notification. I don't think we need to worry about passwords, but we could provide a parameter to disable it if we wanted to mitigate that risk.

**Comment by pmario** [comment]
>I do also like the suggestion of showing a trimmed version of the copied text in the notification.

IMO, there is not enough time to read the trimmed text. 

The OP suggested permalink or permaview ... IMO even those 2 strings have not enough visual difference to "make a difference". 

And even if users see a difference it will be gone way to fast. ... So the next request will be a "Notification history". .. Because if users will miss the info, but see that there was a difference. We will get support requests.

**Comment by Jermolene** [comment]
Notification history is a good idea :smile:

**Comment by yaisog** [comment]
At the risk of drifting a bit into the wrong direction... I have a keyboard shortcut that copies the title of the current tiddler to the clipboard. There seems to be a security mechanism, at least in Firefox, that blocks this when there was no other action for a (short) while, like a mouse click.
Does anyone know if there is a setting to turn this off? In Firefox, not in TW, of course.

**Comment by pmario** [comment]
Then this topic is related to:  **[IDEA] Extend notifications mechanism to make notifications more flexible.** #7040

**Comment by pmario** [comment]
I was thinking about a mechanism that uses the $:/tags/AboveStory area to temporary display notifications. The idea was:

- The area should expand up to 3 messages
  - If there are no messages it should go away
  - It can be switched between "above story" or "below story" ... user configurable
- If there are more than 3 messages oldest scrolls out of view
  - There may be a scrollbar to indicate, that there are more messages
- After some seconds the area shrinks back to 1 message visible. .. the latest
  - If there are "high priority" messages the latest one will be shown 
  - There should be a setting which shows a summary instead of the latest high priority message
  - The area is automatically hidden except if there is a high priority message active
- Every message should get a "close" button
- There are 3 priorities of messages. 
  1. high - eg: Network connection failure ... they need to be removed manually
  2. medium - don't exist yet ... They go away after | more than 1 minute
  3. low - copy to clipboard .. they are deleted automatically after a short period of time | less than 1 minute
- notification priorities are colour coded
- high priority messages can have 5 states
  1. active - not acknowledged ... Message is prominently shown
  2. active - acknowledged ... Message can be hidden but summary is shown .. Summary will not be hidden
  3. inactive - not acknowledged ... Message will be shown or summary ... message will be deleted if "acknowledged" 
  4. inactive - acknowledged ... Message is auto deleted ... no further notification
  5. If message has state 1 and goes to state 4 there should be a visual indiction that the state has changed eg: 3 times blinking or something similar

Just some brainstorming

**Comment by yaisog** [comment]
These are some really nice ideas.
Do we really need active / inactive and so many states?
If sometimes a summary is shown, there should be a way to see the whole message → notification history, maybe?

Maybe we could start by implementing Mohammed's OP wish and multiple priorities. Later, the multiple messages mechanism could build on top of that.

Also, on newer Windows, the "toast" notifications show only one after another. I think on older Windows they could stack, like in your idea. I don't know if they changed it for UX reasons or why...

**Comment by pmario** [comment]
>Do we really need active / inactive and so many states?

Yes, .. As soon as I did acknowledge a "Network connection failure"  I do _not_ want to be annoyed anymore. 

So if I use a Node.js server and I lost connection, this info has "high priority" .. So the info should always be prominently visible **but** only to a degree that TW itself is still usable. ... At the moment that's not the case. The network dialogue "kills" the wiki. 

![image](https://user-images.githubusercontent.com/374655/223946722-61772748-0e0b-420e-a463-41fafafb62a5.png)
